lib/reduce_quda.cu

```diff
@@ -264,487 +311,104 @@ namespace quda {
         param.grid.y = nParity;
       }
 
-      long long flops() const { return arg.r.flops() * x.Length(); }
+      long long flops() const { return r.flops() * x.Length(); }
 
       long long bytes() const
       {
         // the factor two here assumes we are reading and writing to the high precision vector
         // this will evaluate correctly for non-mixed kernels since the +2/-2 will cancel out
-        return (arg.r.streams() - 2) * x.Bytes() + 2 * z.Bytes();
+        return (r.streams() - 2) * x.Bytes() + 2 * z.Bytes();
       }
 
       int tuningIter() const { return 3; }
     };
 
-    template <template <typename ReducerType, typename real> class Reducer, typename real,
-              typename store_t, int len, int N, typename z_store_t = store_t, int Nz = N, typename coeff_t>
-    auto nativeReduce(const coeff_t &a, const coeff_t &b, ColorSpinorField &x, ColorSpinorField &y,
-                      ColorSpinorField &z, ColorSpinorField &w, ColorSpinorField &v, int length)
-    {
-      checkLength(x, y);
-      checkLength(x, z);
-      checkLength(x, w);
-      checkLength(x, v);
-
-      using host_reduce_t = typename Reducer<double, real>::reduce_t;
-      Reducer<device_reduce_t, real> r(a, b);
-
-      Spinor<store_t, N> X(x);
-      Spinor<store_t, N> Y(y);
-      Spinor<z_store_t, Nz> Z(z);
-      Spinor<store_t, N> W(w);
-      Spinor<store_t, N> V(v);
-
-      host_reduce_t value;
-      Reduce<host_reduce_t, real, len, decltype(X), decltype(Y), decltype(Z), decltype(W), decltype(V), decltype(r)>
-        reduce(value, X, Y, Z, W, V, r, x, y, z, w, v, length);
-      reduce.apply(*(blas::getStream()));
-
-      blas::bytes += reduce.bytes();
-      blas::flops += reduce.flops();
-      checkCudaError();
-      return value;
-    }
-
-    /*
-      Wilson
-      double double2 M = 1/12
```

Comment:
These types of `M` values scattered across multiple files confuse the life out of me. Maybe this gets added in a later PR, but if not, can there be some documentation explaining what they mean?

Reply:
Everything was cleared up in a separate convo with @maddyscientist , all is good.